pkg/cmd/plan.go

```diff
@@ -0,0 +1,276 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"github.com/testground/testground/pkg/api"
+	"github.com/testground/testground/pkg/config"
+
+	ttmpl "github.com/testground/plan-templates/templates"
+
+	"github.com/BurntSushi/toml"
+	"github.com/go-git/go-git/v5"
+	gitcfg "github.com/go-git/go-git/v5/config"
+	"github.com/mattn/go-zglob"
+	"github.com/urfave/cli/v2"
+)
+
+var PlanCommand = cli.Command{
+	Name:  "plan",
+	Usage: "plan management",
+	Subcommands: cli.Commands{
+		&cli.Command{
+			Name:  "create",
+			Usage: "create a plan named `PLAN_NAME`",
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:     "remote",
+					Usage:    "origin for your repository i.e. git@github.com:your/repo",
+					Required: false,
+				},
+				&cli.StringFlag{
+					Name:     "target",
+					Usage:    "target language (default: go), avail: [go]",
+					Required: false,
+					Value:    "go",
+				},
+				&cli.StringFlag{
+					Name:  "module",
+					Usage: "module name (used for initial templating",
+					Value: "github.com/your/module/name",
+				},
+				&cli.StringFlag{
+					Name:     "plan",
+					Aliases:  []string{"p"},
+					Usage:    "specifies the name of the plan to create",
+					Required: true,
+				},
+			},
+			Action: createCommand,
+		},
+		&cli.Command{
+			Name:  "import",
+			Usage: "import a plan from the local filesystem or git repository",
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:     "source",
+					Usage:    "specifies the source of the plan to be imported, can be git or local.",
+					Required: true,
+				},
+				&cli.BoolFlag{
+					Name:     "git",
+					Usage:    "use git to import (default: false)",
+					Required: false,
+					Value:    false,
+				},
+				&cli.StringFlag{
+					Name:     "name",
+					Usage:    "override the name of the plan directory (default: automatic)",
+					Required: false,
+				},
+			},
+			Action: importCommand,
+		},
+		&cli.Command{
+			Name:  "rm",
+			Usage: "remove a plan directory from $TESTGROUND_HOME",
+			Flags: []cli.Flag{
+				&cli.BoolFlag{
+					Name:     "yes",
+					Usage:    "confirm removal (without this, the command does nothing)",
+					Required: false,
+					Value:    false,
+				},
+			},
+			Action: rmCommand,
+		},
+		&cli.Command{
+			Name:   "list",
+			Usage:  "enumerate all test cases known to the client",
+			Action: listCommand,
+		},
+	},
+}
+
+// These are the variables used for executing templates during `testground plan create`
+// TODO (cory) Make these specific to the target lang?
+type templateVars struct {
+	Name   string
+	Module string
+}
+
+func createCommand(c *cli.Context) error {
```

Comment:
Did some failure testing:

```
⟩ ./testground plan create --remote git@github.com:raulk/test-plans --plan rauls-plans
this command requires one argument -- specify the plan name
```

The plan name is specified... I would've expected an error message telling me that --target and --module are missing.

Reply:
Even in the happy case it seems to fail with the same error?

```
⟩ ./testground plan create --remote git@github.com:raulk/test-plans --plan rauls-plans --target go --module github.com/raulk/test-plans
this command requires one argument -- specify the plan name
```

Reply:
Definitely a bug!